code/datums/components/radiation_ripple.dm

```diff
@@ -0,0 +1,37 @@
+/* Various forms of radioactive structures, like uranium walls and floors
+ * do not actively emit radiation, unless interacted with. In which case
+ * they then trigger a radiation pulse which then ripples across all
+ * adjacent turfs, which may then activate THOSE to pulse and so on.
+ */
+#define RADIATION_EVENT_COOLDOWN 1.5 SECONDS
+
+/datum/component/radiation_ripple
+	/// Strength of radiation pulse when triggered
+	var/strength
+
+	COOLDOWN_DECLARE(last_event)
+	var/active = FALSE
+
+/datum/component/radiation_ripple/Initialize(signal_or_siglist, strength = 150)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.strength = strength
+
+	RegisterSignal(parent, signal_or_siglist, .proc/trigger)
+	RegisterSignal(parent, COMSIG_RADIATION_RIPPLE_TRIGGER, .proc/trigger)
+
+/datum/component/radiation_ripple/proc/trigger()
+	if(active || !COOLDOWN_FINISHED(src, last_event))
+		return
+
+	active = TRUE
+	radiation_pulse(parent, strength)
+
+	for(var/atom/A in orange(1, parent))
+		SEND_SIGNAL(A, COMSIG_RADIATION_RIPPLE_TRIGGER)
```

Comment:
Using signals like proc calls is kinda icky, they ought to be a representation of an event happening rather then just a hook.
Now I say this, but I'm not sure how you could go about cleaning things up

Reply:
COMSIG_ON_RADIATED thats used by everything the same way

Reply:
wouldn't that break behavior for floor tiles, since they don't actually have enough power to move a tile out from their starting point?

Reply:
reading this on my PC isnt this just exactly what /datum/radiation_wave exists for? It doesn't make a lot of sense to keep these snowflake radiation reactions that are really just remnants of old-rads.